freshmaker/image.py

```diff
@@ -798,8 +798,17 @@ def get_images_by_nvrs(self, nvrs, published=True, content_sets=None,
             ]
 
         def _image_has_rpm(image, rpms):
-            # rpms is a list of rpm names
-            rpm_names = [x["name"] for x in image["rpm_manifest"][0]["rpms"]]
+            try:
+                # rpms is a list of rpm names
+                rpm_names = [x["name"] for x in image["rpm_manifest"][0]["rpms"]]
+            except Exception:
+                # When this is called, the image should always has rpm_manifest, but Pyxis
+                # may has some invalid image data which is missing rpm_manifest, when
+                # such images are returned by Pyxis, it will trigger an exception, we need
+                # to logging the image NVR for troubleshooting purpose.
```

Comment:
I just have a silly linguistic suggestion for the explanatory comment:
```
# When this is called, the image should have rpm_manifest, but Pyxis
# may have some invalid image data which is missing rpm_manifest; when
# such images are returned by Pyxis, it will trigger an exception; we need
# to log the image NVR for troubleshooting purposes.
```
but feel free to ignore it if I didn't understand what you meant!

Reply:
It's much better, I've applied the suggestion, thank you!